src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalCpxSvDiscoverFromLongReadsSpark.java

```diff
@@ -0,0 +1,228 @@
+
+        if (writeSAMFiles) {
+            final Broadcast<SAMFileHeader> headerBroadcast = ctx.broadcast(header);
+            splitUpLongReads.forEach((k, v) -> writeSAM(v, k.name(), reads, headerBroadcast, outputDir, localLogger));
+        }
+
+        new ForSimpleInsDel()
+                .inferSvAndWriteVCF(splitUpLongReads.get(RawTypes.InsDel), outputDir+"/"+RawTypes.InsDel.name()+".vcf",
+                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference, getAuthenticatedGCSOptions(),
+                        localLogger);
+
+//        new ForSimpleStrandSwitch()
+//                .inferSvAndWriteVCF(splitUpLongReads.get(RawTypes.Inv), outputDir+"/"+RawTypes.Inv.name()+".vcf",
+//                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference, getAuthenticatedGCSOptions(),
+//                        localLogger);
+    }
+
+    private enum RawTypes {
+        MultiConf, Inv, InsDel, SegDupOrMEI, Cpx;
+    }
+
+    private static boolean hasEquallyGoodConfigurations(final AlignedContig longread) {
+        return longread.hasEquallyGoodAlnConfigurations;
+    }
+
+    private static boolean hasOnly2Alignments(final AlignedContig longreadWithOnlyOneConfig) {
+        return longreadWithOnlyOneConfig.alignmentIntervals.size() == 2;
+    }
+
+    private static boolean isSameChromosomeMapping(final AlignedContig longreadWithOnlyOneConfigAnd2Aln) {
+        return longreadWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(0).referenceSpan.getContig()
+                .equals(longreadWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(1).referenceSpan.getContig());
+    }
+
+    private static boolean isLikelyInvBreakpointOrInsInv(final AlignedContig longreadWithOnlyOneConfigAnd2AlnToSameChr) {
+        final AlignmentInterval intervalOne = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0),
+                                intervalTwo = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
+        return intervalOne.forwardStrand ^ intervalTwo.forwardStrand;
+    }
+
+    private static boolean isLikelySimpleInsDel(final AlignedContig longreadWithOnlyOneConfigAnd2AlnToSameChr) {
+        if (isLikelyInvBreakpointOrInsInv(longreadWithOnlyOneConfigAnd2AlnToSameChr)) return false;
+
+        final AlignmentInterval intervalOne, intervalTwo;
+        if (longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0).forwardStrand) {
+            intervalOne = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0);
+            intervalTwo = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
+        } else {
+            intervalOne = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
+            intervalTwo = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0);
+        }
+        return ( intervalOne.referenceSpan.getEnd() <= intervalTwo.referenceSpan.getStart() )
```

Comment:
If the first case of this clause is true, won't the second one always be true too? Seems like you could just check the second one..